packages/amplify-category-function/src/provider-utils/awscloudformation/index.ts

```diff
@@ -381,6 +391,31 @@ export async function updateConfigOnEnvInit(context: $TSContext, resourceName: s
     if (currentCloudVersionHash) {
       _.set(amplifyMeta, [categoryName, resourceName, versionHash], currentCloudVersionHash);
     }
+
+    // Since the CFN template and parameters.json are updated on each new layer version which are specific to each env, we need to update
+    // the files accordingly to ensure the correct status is shown after env checkout. The restore flag already handles this scenario.
+    if (context.input.command === 'env' && context.input?.subCommands.includes('checkout') && !context.exeInfo?.inputParams?.restore) {
+      const cfnFilename = `${resourceName}${cfnTemplateSuffix}`;
+      const currentCloudBackendDirPath = pathManager.getCurrentCloudBackendDirPath(projectPath);
+      const currentCfnTemplatePath = path.join(currentCloudBackendDirPath, categoryName, resourceName, cfnFilename);
+      const currentParametersJsonPath = path.join(
```

Comment:
We have `getResourceParametersFilePath` and `getResourceDirectory` (for above) in `PathManager`